File: OOP/1_intro/Reckless+Driver+-+C++/RecklessDriver/RecklessDriver/LetterBox.cpp
#include "LetterBox.h"
#include "Player.h"
#include <iostream>

namespace RecklessDriver {
	LetterBox::LetterBox(int _damage, int _cash):SideObject(_damage, _cash)
	{
		SetName("LetterBox");
	}


	LetterBox::~LetterBox()
	{
	}
	void LetterBox::OnCollision(GameObject *other)
	{
		if (other->GetTag() == "Player") {
			Player *p = (Player *)other;
			if (count == 0) {
				std::cout << "#### COLLISION ->[LetterBox] Letters falling" << std::endl;
				p->ApplyDamage(GetDamage(), GetCash());
			}
			else {
				std::cout << "#### COLLISION ->[LetterBox] Collided again" << std::endl;
				p->ApplyDamage(GetDamage(), GetCash() * count);
			}
		}
	}
}

File: OOP/1_intro/Reckless+Driver+-+Java/Reckless Driver/src/com/poash/rd/LetterBox.java
package com.poash.rd;

public class LetterBox extends SideObject {

	public LetterBox(int damage, int cash) {
		super(damage, cash);
		setName("LetterBox") ;
	}

	@Override
	public void OnCollision(GameObject other) {
		if(other.getTag().equals("Player")){
			Player player = (Player)other; 
			if(count == 0){
				System.out.println("#### COLLISION ->[Letterbox] Letters falling");
				player.ApplyDamage(getDamage(), getCash());
			}else{
				System.out.println("#### COLLISION ->[Letterbox] Collided again");
				player.ApplyDamage(getDamage(), getCash() * count);
			}
		}
	}
}
